File: Julia/math/number_of_digits.jl
# Julia program to Find the Number of Digits in a Number 

# Extracting the number of digits
function number_of_digits(num)
    count = 0
    while(num != 0)
        count = count + 1
        num = num ÷ 10
    end
    return count
end

print("Enter the number: ")
num = readline()
num = parse(Int, num)
count = number_of_digits(abs(num))
println("The number of digits in the given number is $count.")

"""
Time Complexity: O(log(num)), where 'num' is the length of the given number
Space Complexity: O(1)
SAMPLE INPUT AND OUTPUT
SAMPLE 1
Enter the number: -12
The number of digits in the given number is 3.
SAMPLE 2
Enter the number: 43258
The number of digits in the given number is 22.
"""

File: C/math/number_of_digits.c
// C Program to find the Number of Digits of a given number
#include <stdio.h>

int sum_of_digits(int num)
{
    int cnt = 0;
    while (num != 0)
    {
        cnt++;
        // Removing the least significant digit
        num = num / 10;
    }
    return cnt;
}

int main()
{
    int num, cnt;
    printf("\nEnter the number: ");
    scanf("%d", &num);

    // If the number is negative, convert it into positive
    if (num < 0)
    {
        num = num * -1;
    }

    // Call the function
    cnt = sum_of_digits(num);
    // Print the obtained count
    printf("The number of digits in the given number is: %d", cnt);
    return 0;
}

/*
Time Complexity- O(log(num)), where 'num' is the given num
Space Complexity- O(1)

SAMPLE INPUT AND OUTPUT

SAMPLE 1

Enter the number: 1231
The number of digits in the given number is: 4
*/
